test/debug/watch_test.rb

```diff
@@ -49,6 +49,45 @@ def test_debugger_stops_when_the_expression_changes
     end
   end
 
+  class InstanceIVarWatchingTest < TestCase
+    def program
+      <<~RUBY
+        class Student
+          attr_accessor :name
+
+          def initialize(name)
+            @name = name
+            binding.bp(command: "watch @name")
+          end
+        end
+
+        class Teacher
+          attr_accessor :name
+
+          def initialize(name)
+            @name = name
+          end
+        end
+
+        s1 = Student.new("John")
+        t1 = Teacher.new("Jane") # it shouldn't stop for this
+        s1.name = "Josh"
+        "foo"
```

Comment:
Can you add line number to this program?

Reply:
I'd do it later because it's quite painful to modify test content with all the test numbers 🙂 
can have some kind of rake task that can print out test script with line numbers? so it'd be like:

1. write tests  without line numbers
2. run the rake task (or just run `rake test`)
3. it fails tests that have no line numbers & print out the updated version of the scripts
4. developers copy & paste those test scripts

Reply:
> I'd do it later because it's quite painful to modify test content with all the test numbers 🙂
Ok, I got it.

> can have some kind of rake task that can print out test script with line numbers? so it'd be like:
Yeah, I agree with your opinion 😄

Reply:
@ono-max I've been working on the tool set I mentioned (pretty naive though). I will open a PR shortly.

Reply:
Updated